community/detectors/rce/cve20213129/src/main/java/com/google/tsunami/plugins/example/Cve20213129VulnDetector.java

```diff
@@ -0,0 +1,126 @@
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import com.google.tsunami.proto.TextData;
+import com.google.tsunami.proto.AdditionalDetail;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+import com.google.protobuf.ByteString;
+
+@PluginInfo(type = PluginType.VULN_DETECTION, name = "Cve20213129VulnDetector", version = "0.1",
+		description = "This plugin detects Laravel (Version <= 8.4.2) running in debug mode, while "
+				+ "using Ignition (Version <= v2.5.1). "
+				+ "Such instances are vulnerable to an unauthenticated remote code execution vulnerability (CVE-2021-3129), "
+				+ "due to unsafe user input handling.",
+		author = "TSUNAMI_COMMUNITY",
+		bootstrapModule = Cve20213129VulnDetectorBootstrapModule.class)
+
+@ForServiceName({"http", "https"})
+public final class Cve20213129VulnDetector implements VulnDetector {
+
+	private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+	private static final String QUERY_PATH = "_ignition/execute-solution";
+	private static final String QUERY_PAYLOAD = "{\"solution\": \"Facade\\\\Ignition\\\\Solutions\\\\MakeViewVariableOptionalSolution\","
+			+ "\"parameters\": {\"variableName\": \"cve20213129_tsunami\", \"viewFile\": \"phar://Tsunami_iDontExist\"}}";
+	private static final CharSequence DETECTION_STRING = "file_get_contents(phar://Tsunami_iDontExist";
+	
+	private final HttpClient httpClient;
+	private final Clock utcClock;
+
+	@Inject
+	Cve20213129VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+		this.utcClock = checkNotNull(utcClock);
+		this.httpClient = checkNotNull(httpClient);
+	}
+
+	@Override
+	public DetectionReportList detect(TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+		logger.atInfo().log("Cve20213129VulnDetector starts detecting.");
+
+		// An example implementation for a VulnDetector.
+		return DetectionReportList.newBuilder().addAllDetectionReports(matchedServices.stream()
+				.filter(NetworkServiceUtils::isWebService).filter(this::isServiceVulnerable)
+				.map(networkService -> buildDetectionReport(targetInfo, networkService)).collect(toImmutableList()))
+				.build();
+	}
+
+	// Checks whether a given network service is vulnerable. Real detection logic
+	// implemented here.
+	private boolean isServiceVulnerable(NetworkService networkService) {
+		{
+			String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + QUERY_PATH;
+			try {
+				HttpResponse httpResponse = httpClient.send(post(targetUri).setHeaders(HttpHeaders.builder()
+						.addHeader(CONTENT_TYPE, "application/json")
+						.addHeader("Accept", "application/json").build())
```

Comment:
You can import `com.google.common.net.HttpHeaders.ACCEPT` and call `.addHeader(ACCEPT, MediaType.JSON_UTF_8).